api/v1/reviewers/recommendations/ReviewerRecommendationController.php

```diff
@@ -0,0 +1,219 @@
+
+class ReviewerRecommendationController extends PKPBaseController
+{
+    /**
+     * @copydoc \PKP\core\PKPBaseController::getHandlerPath()
+     */
+    public function getHandlerPath(): string
+    {
+        return 'reviewers/recommendations';
+    }
+
+    /**
+     * @copydoc \PKP\core\PKPBaseController::getRouteGroupMiddleware()
+     */
+    public function getRouteGroupMiddleware(): array
+    {
+        return [
+            'has.user',
+            'has.context',
+            self::roleAuthorizer([
+                Role::ROLE_ID_SITE_ADMIN,
+                Role::ROLE_ID_MANAGER,
+                Role::ROLE_ID_SUB_EDITOR,
+            ]),
+        ];
+    }
+
+    /**
+     * @copydoc \PKP\core\PKPBaseController::authorize()
+     */
+    public function authorize(PKPRequest $request, array &$args, array $roleAssignments): bool
+    {
+        $illuminateRequest = $args[0]; /** @var \Illuminate\Http\Request $illuminateRequest */
+        $actionName = static::getRouteActionName($illuminateRequest);
+
+        $this->addPolicy(new UserRolesRequiredPolicy($request), true);
+        $this->addPolicy(new ContextAccessPolicy($request, $roleAssignments));
+
+        if (in_array($actionName, ['edit', 'updateStatus', 'delete'])) {
+            $this->addPolicy(
+                new RecommendationAccessPolicy(
+                    $request,
+                    static::getRequestedRoute($illuminateRequest)->parameter('reviewerRecommendationId')
+                )
+            );
+        }
+
+        return parent::authorize($request, $args, $roleAssignments);
+    }
+
+    /**
+     * @copydoc \PKP\core\PKPBaseController::getGroupRoutes()
+     */
+    public function getGroupRoutes(): void
+    {
+        Route::get('{reviewerRecommendationId}', $this->get(...))
+            ->name('reviewer.recommendations.get')
+            ->whereNumber(['reviewerRecommendationId']);
+
+        Route::get('', $this->getMany(...))
+            ->name('reviewer.recommendations.getMany');
+
+        Route::post('', $this->add(...))
+            ->name('reviewer.recommendations.add');
+
+        Route::put('{reviewerRecommendationId}', $this->edit(...))
+            ->name('reviewer.recommendations.edit')
+            ->whereNumber(['reviewerRecommendationId']);
+
+        Route::put('{reviewerRecommendationId}/status', $this->updateStatus(...))
+            ->name('reviewer.recommendations.edit.status')
+            ->whereNumber(['reviewerRecommendationId']);
+
+        Route::delete('{reviewerRecommendationId}', $this->delete(...))
+            ->name('reviewer.recommendations.delete')
+            ->whereNumber(['reviewerRecommendationId']);
+    }
+
+    /**
+     * Get specific recommendation response
+     */
+    public function get(Request $illuminateRequest): JsonResponse
+    {
+        $recommendation = ReviewerRecommendation::find($illuminateRequest->route('reviewerRecommendationId'));
```

Comment:
This should account for a recommendation not being found, returning a 404 if none is found.

Reply:
RIght , this should get included in `in_array($actionName, ['get', 'edit', 'updateStatus', 'delete'])`. Added it .